### PR TITLE
Tide history page contains correct links for Gerrit

### PR DESCRIPTION
**Pull request by chaodaiG**
These values were hardcoded to the GitHub url pattern

### CODE REVIEW

File: prow/cmd/deck/main.go [hunk 1]
```diff
@@ -589,6 +591,7 @@ func prodOnlyMain(cfg config.Getter, pluginAgent *plugins.ConfigAgent, authCfgGe
 
 	// Handles link to github
 	mux.HandleFunc("/github-link", HandleGitHubLink(o.github.Host, secure))
+	mux.HandleFunc("generic-provider-link", HandleGenericProviderLink(o.github.Host, secure))
```
Comment (spiffxp) [inline prow/cmd/deck/main.go:594]: Make consistent with `/github-link` above?
```suggestion
	mux.HandleFunc("/generic-provider-link", HandleGenericProviderLink(o.github.Host, secure))
```

File: prow/cmd/deck/main.go [hunk 2]
```diff
@@ -232,6 +233,7 @@ var simplifier = simplifypath.NewSimplifier(l("", // shadow element mimicing the
 	l("github-login",
 		l("redirect")),
 	l("github-link"),
+	l("generic-provider-link"),
```
Comment (spiffxp) [inline prow/cmd/deck/main.go:236]: Naming is hard.

"provider" is a very overloaded term for me, is this how you commonly refer to "github or gerrit" within prow?

Wonder if we could be more specific like `/git-provider-link`, `/git-review-link`, `/git-ui-link`, `/scm-link`

File: prow/cmd/deck/static/tide-history/tide-history.ts [hunk 3]
```diff
@@ -36,12 +37,15 @@ function optionsForRepoBranch(repo: string, branch: string): Options {
   const hist: {[key: string]: Record[]} = typeof tideHistory !== 'undefined' ? tideHistory.History : {};
   const poolKeys = Object.keys(hist);
   for (const poolKey of poolKeys) {
-    const match = RegExp('(.*?):(.*)').exec(poolKey);
+    // poolKey is in the formatted as `<org>/<repo>:branch`, for example:
+    // - GitHub: `foo/bar:main` # repo: bar
+    // - Gerrit: `https://foo/bar/baz:main` # repo: bar/baz
+    const match = RegExp('(((https|http)://)?.*?):(.*)').exec(poolKey);
```
Comment (spiffxp) [inline prow/cmd/deck/static/tide-history/tide-history.ts:43]: I think the existing regex would get `repo: foo/bar/baz` so either the comment or the regex is wrong?
```suggestion
    // - Gerrit: `https://foo/bar/baz:main` # repo: bar/baz
    const match = RegExp('(((https|http)://[^/]*/)?.*?):(.*)').exec(poolKey);
```

File: prow/cmd/deck/static/common/common.ts [hunk 4]
```diff
@@ -154,7 +154,7 @@ export namespace cell {
       if (pull.author_link) {
         al.href = pull.author_link;
       } else {
-        al.href = `/github-link?dest=${  pull.author}`;
+        al.href = `/generic-provider-link?target=author&author=${  pull.author}`;
```
Comment (spiffxp) [inline prow/cmd/deck/static/common/common.ts:157]:
```suggestion
        al.href = `/generic-provider-link?target=author&author=${pull.author}`;
```

File: prow/cmd/deck/main.go [hunk 5]
```diff
@@ -1495,6 +1498,52 @@ func HandleGitHubLink(githubHost string, secure bool) http.HandlerFunc {
 	}
 }
 
+// HandleGenericProviderLink returns link based on different providers.
+func HandleGenericProviderLink(githubHost string, secure bool) http.HandlerFunc {
```
Comment (spiffxp) [inline prow/cmd/deck/main.go:1502]: Please add `TestHandleGenericProviderLink` to main_test.go

**Review by spiffxp — CHANGES_REQUESTED** [review]
Makes sense, needs more test coverage though

Is there a reason we don't want to replace/remove `/github-link`?

File: prow/cmd/deck/main.go (hunk 2 above)
Comment (chaodaiG) [reply prow/cmd/deck/main.go]: agreed and picked `git-provider-link`

File: prow/cmd/deck/main.go (hunk 1 above)
Comment (chaodaiG) [reply prow/cmd/deck/main.go]: done

File: prow/cmd/deck/main.go (hunk 5 above)
Comment (chaodaiG) [reply prow/cmd/deck/main.go:1502]: done

File: prow/cmd/deck/static/common/common.ts (hunk 4 above)
Comment (chaodaiG) [reply prow/cmd/deck/static/common/common.ts]: done

File: prow/cmd/deck/static/tide-history/tide-history.ts (hunk 3 above)
Comment (chaodaiG) [reply prow/cmd/deck/static/tide-history/tide-history.ts]: done.

**Review by chaodaiG — COMMENTED** [review]
/hold
Will unhold once dryrun confirmed that it works

**Comment by chaodaiG** [comment]
Dry-run looks good to me, @spiffxp please take another look
/unhold

File: prow/cmd/deck/main.go [hunk 6]
```diff
@@ -1495,6 +1498,54 @@ func HandleGitHubLink(githubHost string, secure bool) http.HandlerFunc {
...
```
Comment (spiffxp) [inline prow/cmd/deck/main.go:1502]:
```suggestion
func HandleGitProviderLink(githubHost string, secure bool) http.HandlerFunc {
```

File: prow/cmd/deck/main_test.go [hunk 7]
```diff
@@ -992,6 +992,94 @@ func TestHandleGitHubLink(t *testing.T) {
 	}
 }
 
+func TestHandleGenericProviderLink(t *testing.T) {
```
Comment (spiffxp) [inline prow/cmd/deck/main_test.go:995]:
```suggestion
func TestHandleGitProviderLink(t *testing.T) {
```

**Review by spiffxp — CHANGES_REQUESTED** [review]
LGTM, just missed renaming the handler function

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Pull request merged**